ds3-sdk/src/main/java/com/spectralogic/ds3client/networking/NetworkClientDecorator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3client.networking;
+
+import com.spectralogic.ds3client.commands.interfaces.Ds3Request;
+
+import java.io.IOException;
+
+public abstract class NetworkClientDecorator implements NetworkClient {
```

Comment:
Some javadoc should be added to explain what this decorator is doing.  From inspection it looks like this is a base decorator so that implementers of other decorators only have to implement the specific functions that they want to modify.